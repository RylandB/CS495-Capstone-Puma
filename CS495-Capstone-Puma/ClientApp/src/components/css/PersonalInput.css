.materialize-input{
    text-align: center;
}

body {
    text-align: center;
    margin-left: 10%;
    margin-right: 10%;
}

form{
    text-align: center;
}

h3{
    font-family: Calibri;
}