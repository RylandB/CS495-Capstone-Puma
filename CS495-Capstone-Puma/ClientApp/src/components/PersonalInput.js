import React from 'react';
import  './css/PersonalInput.css'
import M from 'materialize-css'

let state ={
    inputFirstName: null,
    inputMiddleName: null,
    inputLastName: null,
    inputHonorific: null,
    inputEmailAddress:null
};

export class PersonalInput extends React.Component{
    componentDidMount(){
        console.log("component did mount");
        M.AutoInit();
    }

    constructor(props){
        super(props);
        this.state = state;
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    componentWillUnmount() {
        state = this.state;
    }

    async handleInputChange(event){

        const target = event.target;
        const value = event.target.value;
        const name = target.name;

        await this.setState({
            [name]: value
        });
        this.props.personalCallback(this.state.inputFirstName, this.state.inputMiddleName, this.state.inputLastName, this.state.inputHonorific, this.state.inputEmailAddress);
    }

    render(){
        return(
            <div className="container">
                <div className="row">
                    <div className="col s6 offset-s3">
                    <h3>Input</h3>
                    <br />
                    <div>
                        <div className = "input-field">
                            <label>Asset Code</label>
                            <input type="text" name="inputFirstName" onChange={this.handleInputChange} value={this.state.inputFirstName}/>
                        </div>
                    </div>

                    <div>
                        <div className = "input-field">
                            <label>Symbol</label>
                            <input type="text" name="inputMiddleName" onChange={this.handleInputChange} value={this.state.inputMiddleName}/>
                        </div>
                    </div>

                    <div>
                        <div className = "input-field">
                            <label>Issue</label>
                            <input type="text" name="inputLastName" onChange={this.handleInputChange} value={this.state.inputLastName}/>
                        </div>
                    </div>

                    <div>
                        <div className = "input-field">
                            <label>Issuer</label>
                            <input type="text" name="inputHonorific" onChange={this.handleInputChange}  value={this.state.inputHonorific} />
                        </div>
                    </div>

                    <div>
                        <div className = "input-field">
                            <label>Quantity</label>
                            <input type="number" name="inputEmailAddress" onChange={this.handleInputChange} value={this.state.inputEmailAddress}/>
                        </div>
                    </div>
                    </div>
                </div>
            </div>
        );
    }

}