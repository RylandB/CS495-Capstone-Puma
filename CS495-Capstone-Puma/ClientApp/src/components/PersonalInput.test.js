import React from 'react';
import {PersonalInput} from "./PersonalInput";

it("PersonalInput Renders Correctly", () => {
    const wrapper = shallow(
        <PersonalInput />
    );

    expect(wrapper).toMatchSnapshot();

});