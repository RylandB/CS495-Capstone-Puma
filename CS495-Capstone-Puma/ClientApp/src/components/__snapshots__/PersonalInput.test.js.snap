// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInput Renders Correctly 1`] = `
<div>
  <h3>
    Account Holder Information
  </h3>
  <br />
  <label>
    First Name
  </label>
  <br />
  <input
    name="inputFirstName"
    onChange={[Function]}
    type="text"
    value={null}
  />
  <br />
  <br />
  <label>
    Middle Name
  </label>
  <br />
  <input
    name="inputMiddleName"
    onChange={[Function]}
    type="text"
    value={null}
  />
  <br />
  <br />
  <label>
    Last Name
  </label>
  <br />
  <input
    name="inputLastName"
    onChange={[Function]}
    type="text"
    value={null}
  />
  <br />
  <br />
  <label>
    Honorific
  </label>
  <br />
  <select
    name="inputHonorific"
    onChange={[Function]}
    value={null}
  >
    <option
      value="Mr"
    >
      Mr
    </option>
    <option
      value="Mrs"
    >
      Mrs
    </option>
    <option
      value="Ms"
    >
      Ms
    </option>
    <option
      value="Dr"
    >
      Dr
    </option>
    <option
      selected="selected"
      value=""
    >
       
    </option>
  </select>
  <br />
  <br />
  <label>
    Email
  </label>
  <br />
  <input
    name="inputEmailAddress"
    onChange={[Function]}
    type="text"
    value={null}
  />
  <br />
  <br />
  <hr />
</div>
`;
